cpmpy/expressions/core.py

```diff
@@ -166,26 +166,19 @@ def has_subexpr(self):
         if hasattr(self, '_has_subexpr'):
             return self._has_subexpr
 
-        # Initialize stack with args
-        stack = list(self.args)
-
-        while stack:
-            el = stack.pop()
-            if isinstance(el, Expression):
```

Comment:
we still need the explicit check that we are dealing with a cpmpy expression, this new interpration will see a constant '3' as a subexpression now since it's not any of our leaf expressions